c7n/resources/awslambda.py

```diff
@@ -644,10 +644,13 @@ class LayerCrossAccount(CrossAccountAccessFilter):
     def process(self, resources, event=None):
         client = local_session(self.manager.session_factory).client('lambda')
         for r in resources:
-            r['c7n:Policy'] = self.manager.retry(
-                client.get_layer_version_policy,
-                LayerName=r['LayerName'],
-                VersionNumber=r['Version']).get('Policy')
+            if 'c7n:Policy' not in r:
+                rpolicy = self.manager.retry(
+                    client.get_layer_version_policy,
+                    LayerName=r['LayerName'],
+                    VersionNumber=r['Version']).get('Policy')
+                if rpolicy:
```

Comment:
i think you want the attribute set even in the negative case, else you'll have low cache utilization.

Reply:
Sorry, wasn't able to follow you here. Are you asking to set `rpolicy = {}` in negative cases?

Reply:
I think I addressed this, but feel free to correct if I'm wrong 😄 . Thanks